Stock.Api/Controllers/ProviderController.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Stock.Api.DTOs;
+using Stock.Api.Extensions;
+using Stock.AppService.Services;
+using Stock.Model.Entities;
+
+namespace Stock.Api.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/provider")]
+    [ApiController]
+    public class ProviderController : ControllerBase
+    {
+        private ProviderService service;
+        private readonly IMapper mapper;
+
+        public ProviderController(ProviderService service, IMapper mapper)
+        {
+            this.service = service;
+            this.mapper = mapper;
+        }
+
+        [HttpPost]
+        public ActionResult Post([FromBody] ProviderDTO value)
+        {
+            TryValidateModel(value);
+
+            try
+            {
+                var provider = this.mapper.Map<Provider>(value);
+                this.service.Create(provider);
+                value.Id = provider.Id;
+                return Ok(new { Success = true, Message = "", data = value });
+            }
+            catch
```

Comment:
Faltaria agregar la excepcion en el catch: catch (TipoExcepcion ex). De todos modos si tiras una excepcion generica y la atrapas aca devolviendo solo "The name is alredy in use", que pasaria si el error fue otro? seria un poco confuso. Ver de generar User-Defined exceptions para el caso en particular de nombre duplicado, e investigar y ver como atrapar excepciones diferentes en un mismo catch.